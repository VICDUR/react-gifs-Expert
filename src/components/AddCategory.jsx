import { useState } from "react";
import Swal from 'sweetalert2';

export const AddCategory = ({ onAddCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue( event.target.value );
    };

    const onSubmit = (event) => {
        event.preventDefault();
        
        if(inputValue.trim().length <= 1 ) return Swal.fire({
            title: 'UPSS!',
            text: 'No hay suficiente texto para buscar.',
            imageUrl: 'https://media4.giphy.com/media/U2nN0ridM4lXy/giphy.gif?cid=790b7611d73b3aa850f47282d8e972bf3a473bdc318ebd81&rid=giphy.gif&ct=g',
            imageWidth: 300,
            imageHeight: 200,
            imageAlt: 'Custom image',
            buttons: false, 
            timer: 2500
          })          

        setInputValue('');
        onAddCategory(inputValue.trim());
    }

  return (
    <form onSubmit={ onSubmit }>
        <input 
            type="text"
            placeholder="Buscar gifs"
            value={inputValue}
            onChange={ onInputChange }
        />
    </form>
  )
};
