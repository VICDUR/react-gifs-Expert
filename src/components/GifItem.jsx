

export const GifItem = ({url, title, id}) => {
  
  const titleModified = title.slice(0, 10).trim();

  return (
    <div className='card'>
        <img src={ url } alt={ title } />
        <p> { titleModified } </p>
    </div>
  )
}
