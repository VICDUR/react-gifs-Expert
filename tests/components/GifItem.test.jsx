import { render, screen } from "@testing-library/react";
import { GifItem } from '../../src/components/GifItem';

// console.log('render', render)


describe('Prueba en Gifitem', () => { 

    const url = "https://superman.com/clarken.jpg";
    const title = "supermanVolador";

    test('debe de hacer match con el snapshot', () => { 

        const { container } = render( <GifItem url={url} title={title} />);
        expect( container ).toMatchSnapshot();

    })


    test('debe de mostrar la imagen con el url y alt indicado', () => { 

        render( <GifItem url={url} title={title} />)
        // screen.debug()
        // expect( screen.getByRole('img').src).toBe(url);
        // expect( screen.getByRole('img').alt).toBe(title);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    })

    test('debe de mostrar el titulo con un largo de 10 caracteres', () => { 

        render( <GifItem url={url} title={title} />);
        expect( screen.getByText( title.slice(0, 10).trim() )).toBeTruthy();

     

    })


 });