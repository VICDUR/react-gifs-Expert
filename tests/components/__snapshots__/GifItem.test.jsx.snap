// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en Gifitem debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="supermanVolador"
      src="https://superman.com/clarken.jpg"
    />
    <p>
       
      supermanVo
       
    </p>
  </div>
</div>
`;
